android/src/main/java/io/keen/client/android/AndroidJsonHandler.java

```diff
@@ -93,9 +137,25 @@ public void setWrapNestedMapsAndCollections(boolean value) {
      */
     private boolean isWrapNestedMapsAndCollections = (Build.VERSION.SDK_INT < 19);
 
+    /** Stringifier for converting JSONObjects to Strings; used for unit tests. */
+    private JsonObjectManager jsonObjectManager = null;
+
     ///// PRIVATE METHODS /////
 
     /**
+     * Get the default jsonObjectManager, or use one that was explicitly specified.
+     *
+     * @return A default implementation which uses the normal Android library methods, unless a
+     * different implementation has been set explicitly.
+     */
+    private JsonObjectManager getJsonObjectManager() {
```

Comment:
question: is it possible this is called from multiple threads? if so, do we need to worry about synchronization?